test/test_fit_som.py

```diff
@@ -0,0 +1,145 @@
+import hypothesis.strategies as hst
+import matplotlib.pyplot as plt
+import numpy as np
+import pytest
+from hypothesis import assume, given, HealthCheck, settings, Verbosity
+from hypothesis.strategies import composite
+from numpy.random import default_rng
+
+from PyDynamic import fit_som, make_semiposdef, sos_FreqResp
+from PyDynamic.examples.demonstrate_fit_som import (
+    demonstrate_second_order_model_fitting,
+)
+from .conftest import hypothesis_bounded_float, hypothesis_float_vector
+
+
+def test_demonstrate_second_order_model_fitting(monkeypatch):
+    # With this expression we override the matplotlib.pyplot.show method with a
+    # lambda expression returning None but only for this one test.
+    # guarantee this
+    monkeypatch.setattr(plt, "show", lambda: None, raising=True)
+    demonstrate_second_order_model_fitting(runs=100)
+
+
+@composite
+def random_input_to_fit_som(draw, guarantee_UH_as_matrix: bool = False):
+    rng = default_rng()
+    # sensor/measurement system
+    S0 = draw(hypothesis_bounded_float(min_value=0.124, max_value=0.124))
+    uS0 = draw(hypothesis_bounded_float(min_value=1e-4, max_value=1e-4))
+    delta = draw(hypothesis_bounded_float(min_value=1e-2, max_value=1e-2))
+    udelta = draw(hypothesis_bounded_float(min_value=1e-3, max_value=1e-3))
+    f0 = draw(hst.integers(min_value=36, max_value=36))
+    uf0 = draw(hypothesis_bounded_float(min_value=0.5, max_value=0.5))
+
+    # Monte Carlo for calculation of unc. assoc. with [real(H),imag(H)]
+    MCruns = draw(hst.sampled_from((10, 20, 40)))
+    white_noise_S0s = rng.normal(loc=S0, scale=uS0, size=MCruns)
+    white_noise_deltas = rng.normal(loc=delta, scale=udelta, size=MCruns)
+    white_noise_f0s = rng.normal(loc=f0, scale=uf0, size=MCruns)
+    frequencies = np.linspace(0, 1.2 * f0, 30)
```

Comment:
Can we wiggle these inputs to really check more randomly.

Reply:
No, we cannot that simple. Small wiggling results in failing tests. This should be done more educated and thus is postponed. At least, there is some randomness in the noise we draw for the Monte Carlo process.

Reply:
Still we replaced the "random" draws by constant values from the example in 9b06699.